source/slang/ir.cpp

```diff
@@ -1495,35 +1490,24 @@ namespace Slang
             }
         }
 
-        IRInstKey key;
-        key.inst = inst;
-
-        // Ideally we would add if not found, else return if was found instead of testing & then adding.
-        IRInst* foundInst = nullptr;
-        bool found = builder->sharedBuilder->globalValueNumberingMap.TryGetValue(key, foundInst);
-
-        SLANG_ASSERT(endCursor == memoryArena.getCursor());
-        
-        if (found)
+        // Find or add the key/inst
         {
-            memoryArena.rewindToCursor(cursor);
-            return foundInst;
-        }
+            IRInstKey key{ inst };
```

Comment:
I can't remember when that initialization syntax came into C++. Is it a C++11 thing or a C++14 thing? No matter what, we should try to stick with C++11 compatibility for as long as possible.

Is there a reason to use that syntax over the more traditional C-like `IRInstKey key = { inst };` or the equally reasonable `IRInstKey key = IRInstKey(inst)` if there is a constructor? The C++ constructor-initialization syntax is kind of awful for readability, so I'd prefer to avoid it when possible.